src/librawspeed/io/Buffer.h

```diff
@@ -22,15 +22,19 @@
 #pragma once
 
 #include "common/Common.h" // for uchar8, uint32, uint64
+#include "io/IOException.h"// for ThrowIOE
 #include "io/Endianness.h" // for getByteSwapped
 #include <algorithm>       // for swap
 
 namespace RawSpeed {
 
-// All file maps have this much space extra at the end. This is useful for
-// BitPump* that needs to have a bit of extra space at the end to be able to
-// do more efficient reads without crashing
-#define FILEMAP_MARGIN 16UL
+// This allows to specify the nuber of bytes that each Buffer needs to
+// allocate additionally to be able to remove one runtime bounds check
+// in BitSream::fill. There are two options:
+// 0 : allocate exactly as much data as required
+// 4 : add minimum number of bytes to keep all BitStream implementations happy
+//     this disables the bounds checking, saves about 1% on modern CPUs
+#define BUFFER_PADDING 0
```

Comment:
> Is there any chance that RawSpeed will work with mmap-ed file? No extra bytes at end, sure.

Yes, after this is merged, that will be possible. Actually, it would have been almost certainly been possible even without this due to block/page sizes involved.

Note though: there is (disabled) code inside the `FileReader` that uses memory mapping and it was disabled (presumably) because it actually makes the code slower. That is also what my personal experience showed in similar situations.

> i will be fixing that

Is that so? This is _exactly_ what _this_ pr fixes. (My pr sets the padding to `0`).